src/Datadog.Trace/HttpOverStreams/DatadogHttpClient.cs

```diff
@@ -87,15 +128,12 @@ bool IsNewLine()
             }
 
             // Skip to status code
-            while (streamPosition < statusCodeStart)
-            {
-                GoNextChar();
-            }
+            await SkipUntil(statusCodeStart).ConfigureAwait(false);
 
             // Read status code
             while (streamPosition < statusCodeEnd)
             {
-                GoNextChar();
+                await GoNextChar().ConfigureAwait(false);
```

Comment:
If we know that the status code is always 3 chars/bytes, should we read read 3 bytes from the stream instead of one at a time?

Reply:
I don't think it's worth optimising this specific case now. We should have a proper buffered reader at some point, which would handle the general case

Reply:
Agreed. Resolving this.